### DISCUSSION TITLE
addMotifAnnotations - parallel computing

### DISCUSSION THREAD

**Discussion by brianpenghe** [GreenleafLab/ArchR#1136] [Questions / Documentation]
I'm running addMotifAnnotations and it's quite slow.
Do you know how to run it with multiple threads?

Thanks!

**Comment by rcorces** [comment]
it is not parallelized so there is no threads argument. all of the work is being done by motifmatchr

**Reply by brianpenghe** [reply]
OK. Is there any chance to have this run in parallel?

**Reply by rcorces** [reply]
If NIH gives us money to hire a dedicated software developer :)

That being said, this really should not be that slow unless you are doing something unconventional.

**Reply by brianpenghe** [reply]
Very well said. NIH is petty and conservative indeed. The current version of ArchR is already quite good to be honest. I've also suggested to the Scanpy team to learn from the cluster ordering that ArchR does.
